library/azure_rm_cosmosdbaccount.py

```diff
@@ -0,0 +1,488 @@
+                    },
+                    'database_account_offer_type': 'default',
+                    'ip_range_filter': 'default',
+                    'is_virtual_network_filter_enabled': 'default',
+                    'enable_automatic_failover': 'default',
+                    'capabilities': {
+                        'name': 'index'
+                    },
+                    'virtual_network_rules': {
+                        'id': 'index'
+                    },
+                    'enable_multiple_write_locations': 'default'
+                })):
+                    self.to_do = Actions.Update
+
+        if (self.to_do == Actions.Create) or (self.to_do == Actions.Update):
+            self.log("Need to Create / Update the Database Account instance")
+
+            if self.check_mode:
+                self.results['changed'] = True
+                return self.results
+
+            response = self.create_update_databaseaccount()
+
+            self.results['changed'] = True
+            self.log("Creation / Update done")
+        elif self.to_do == Actions.Delete:
+            self.log("Database Account instance deleted")
+            self.results['changed'] = True
+
+            if self.check_mode:
+                return self.results
+
+            self.delete_databaseaccount()
+            # make sure instance is actually deleted, for some Azure resources, instance is hanging around
+            # for some time after deletion -- this should be really fixed in Azure.
+            while self.get_databaseaccount():
+                time.sleep(20)
+        else:
+            self.log("Database Account instance unchanged")
+            self.results['changed'] = False
+            response = old_response
+
+        if self.state == 'present':
+            self.results.update(self.format_item(response))
+        return self.results
+
+    def create_update_databaseaccount(self):
+        '''
+        Creates or updates Database Account with the specified configuration.
+
+        :return: deserialized Database Account instance state dictionary
+        '''
+        self.log("Creating / Updating the Database Account instance {0}".format(self.name))
+
+        try:
+            response = self.mgmt_client.database_accounts.create_or_update(resource_group_name=self.resource_group,
+                                                                           account_name=self.name,
+                                                                           create_update_parameters=self.parameters)
+            if isinstance(response, LROPoller) or isinstance(response, AzureOperationPoller):
+                response = self.get_poller_result(response)
+
+        except CloudError as exc:
+            self.log('Error attempting to create the Database Account instance.')
+            self.fail("Error creating the Database Account instance: {0}".format(str(exc)))
+        return response.as_dict()
```

Comment:
again, pls not use as_dict(). in webapp module, when call as_dict() on appservice plan, sometimes get 'id'  not  valid key error. after do maunal serialization, the error  never happens.

Reply:
well, if as_dict doesn't work it's some problem with sdk, and probably it happens only for webapp, it should be fixed by the sdk.